ambry-router/src/test/java/com.github.ambry.router/GetOperationTest.java

```diff
@@ -0,0 +1,474 @@
+        operationCallback, time);
+
+    Assert.assertEquals("Callback must match", operationCallback, op.getCallback());
+    Assert.assertEquals("Futures must match", operationFuture, op.getFuture());
+    Assert.assertEquals("Blob ids must match", blobIdStr, op.getBlobIdStr());
+    // count was incremented, but this operation was not completed, so decrement.
+    NonBlockingRouter.decrementOperationsCount();
+  }
+
+  /**
+   * Test basic successful operation completion, by polling and handing over responses to the BlobInfo operation.
+   * @throws Exception
+   */
+  @Test
+  public void testPollAndResponseHandling()
+      throws Exception {
+    GetBlobInfoOperation op =
+        new GetBlobInfoOperation(routerConfig, mockClusterMap, responseHandler, blobIdStr, operationFuture,
+            operationCallback, time);
+    ArrayList<RequestInfo> requestListToFill = new ArrayList<>();
+    requestRegistrationCallback.requestListToFill = requestListToFill;
+    op.poll(requestRegistrationCallback);
+    Assert.assertEquals("There should only be as many requests at this point as requestParallelism", requestParallelism,
+        correlationIdToGetOperation.size());
+
+    List<ResponseInfo> responses = sendAndWaitForResponses(requestListToFill);
+    for (ResponseInfo response : responses) {
+      op.handleResponse(response);
+      if (op.isOperationComplete()) {
+        break;
+      }
+    }
+    Assert.assertTrue("Operation should be complete at this time", op.isOperationComplete());
+    assertSuccess();
+  }
+
+  /**
+   * Test the case where all requests time out within the GetOperation.
+   * @throws Exception
+   */
+  @Test
+  public void testRouterRequestTimeoutAllFailure()
+      throws Exception {
+    GetBlobInfoOperation op =
+        new GetBlobInfoOperation(routerConfig, mockClusterMap, responseHandler, blobIdStr, operationFuture,
+            operationCallback, time);
+    requestRegistrationCallback.requestListToFill = new ArrayList<>();
+    op.poll(requestRegistrationCallback);
+    while (!op.isOperationComplete()) {
+      time.sleep(routerConfig.routerRequestTimeoutMs + 1);
+      op.poll(requestRegistrationCallback);
+    }
+    // At this time requests would have been created for all replicas, as none of them were delivered,
+    // and cross-colo proxying is enabled by default.
+    Assert.assertEquals("Must have attempted sending requests to all replicas", replicasCount,
+        correlationIdToGetOperation.size());
+    Assert.assertTrue("Operation should be complete at this time", op.isOperationComplete());
+    Assert.assertTrue(operationFuture.isDone());
+    Assert.assertEquals(operationException.get(), op.getOperationException());
+    RouterException routerException = (RouterException) operationException.get();
+    Assert.assertEquals(RouterErrorCode.OperationTimedOut, routerException.getErrorCode());
```

Comment:
changed this to move the operation completing part outside this unit and into the `GetManager`. This simplified the testing and was consistent with the `PutManager`. Therefore neither the future nor the callback will be checked in the tests.